add style to list separator, so one could write to userstyle.css:

table.dataplugin_table span.sep {
    display: block;
    font-size: 0px;
}

and have list items listed below each other

--- dokuwiki-plugin-data-20160311/helper.php~	2016-07-01 14:27:21.000000000 +0300
+++ dokuwiki-plugin-data-20160311/helper.php	2016-11-22 23:05:37.213943088 +0200
@@ -360,7 +360,7 @@
                     }
             }
         }
-        return join(', ', $outs);
+        return join('<span class="sep">, </span>',$outs);
     }
 
     /**
